api/config/config.go

```diff
@@ -264,6 +265,32 @@ func (cm *ConfigLoader) ListConfigs() []string {
 	return res
 }
 
+func (cm *ConfigLoader) Preload(modelPath string) error {
+	cm.Lock()
+	defer cm.Unlock()
+
+	for i, config := range cm.configs {
+		if strings.Contains(config.PredictionOptions.Model, "http://") || strings.Contains(config.PredictionOptions.Model, "https://") {
+			// md5 of model name
+			md5Name := utils.MD5(config.PredictionOptions.Model)
+
+			// check if file exists
+			if _, err := os.Stat(filepath.Join(modelPath, md5Name)); err == os.ErrNotExist {
+				err := utils.DownloadFile(config.PredictionOptions.Model, filepath.Join(modelPath, md5Name))
```

Comment:
TODO: this is a dup, we have already model galleries downloading, so it should re-use the same utilities (also because it shows the progress, too)